app/repositories/measurements_repository.rb

```diff
@@ -3,15 +3,13 @@ def from_last_24_hours(stream_id:)
     Measurement.where(stream_id: stream_id).reorder(time: :desc).limit(1440)
   end
 
-  def stream_daily_average_value(stream_id:, beginning_of_day:)
-    end_of_day = beginning_of_day + 1.day
-
+  def stream_daily_average_value(stream_id:, time_with_time_zone:)
     Measurement
       .where(
         'stream_id = ? AND time_with_time_zone >= ? AND time_with_time_zone < ?',
```

Comment:
My bad, I've missed it - as we use `end_of_day` right now, we should have `<=` in the last condition:
`stream_id = ? AND time_with_time_zone >= ? AND time_with_time_zone <= ?`